clustered/server/src/main/java/org/ehcache/clustered/server/management/Management.java

```diff
@@ -127,26 +129,31 @@ public void clientDisconnected(ClientDescriptor clientDescriptor, ClientState cl
   public void clientReconnected(ClientDescriptor clientDescriptor, ClientState clientState) {
     if (managementRegistry != null) {
       LOGGER.trace("clientReconnected({})", clientDescriptor);
-      managementRegistry.refresh();
+      managementRegistry.refresh(); // required because ClientState fields have been modified
       managementRegistry.pushServerEntityNotification(new ClientStateBinding(clientDescriptor, clientState), EHCACHE_CLIENT_RECONNECTED.name());
     }
   }
 
   public void sharedPoolsConfigured() {
     if (managementRegistry != null) {
       LOGGER.trace("sharedPoolsConfigured()");
-      ehcacheStateService.getSharedResourcePools()
+      CompletableFuture.allOf(ehcacheStateService.getSharedResourcePools()
         .entrySet()
-        .forEach(e -> managementRegistry.register(new PoolBinding(e.getKey(), e.getValue(), PoolBinding.AllocationType.SHARED)));
-      managementRegistry.refresh();
-      managementRegistry.pushServerEntityNotification(PoolBinding.ALL_SHARED, EHCACHE_RESOURCE_POOLS_CONFIGURED.name());
+        .stream()
+        .map(e -> managementRegistry.register(new PoolBinding(e.getKey(), e.getValue(), PoolBinding.AllocationType.SHARED)))
+        .collect(Collectors.toList())
+        .toArray(new CompletableFuture<?>[0]))
```

Comment:
this line and the one above can be collapsed in
`.toArray(CompletableFuture[]::new)`